src/Features/Core/Portable/Formatting/AbstractNewDocumentFormattingService.cs

```diff
@@ -0,0 +1,40 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.ErrorReporting;
+
+namespace Microsoft.CodeAnalysis.Formatting
+{
+    internal abstract class AbstractNewDocumentFormattingService : INewDocumentFormattingService
+    {
+        private readonly IEnumerable<INewDocumentFormattingProvider> _providers;
+
+        protected AbstractNewDocumentFormattingService(IEnumerable<INewDocumentFormattingProvider> providers)
+        {
+            _providers = providers;
+        }
+
+        public async Task<Document> FormatNewDocumentAsync(Document document, CancellationToken cancellationToken)
+        {
+            foreach (var provider in _providers)
+            {
+                cancellationToken.ThrowIfCancellationRequested();
+
+                try
+                {
+                    document = await provider.FormatNewDocumentAsync(document, cancellationToken).ConfigureAwait(false);
+                }
+                catch (Exception ex) when (FatalError.ReportAndCatchUnlessCanceled(ex, cancellationToken))
```

Comment:
should we break out at this point?

Reply:
I don't feel too strongly about it, but if a single formatter has a bug I don't know why we wouldn't keep using the rest, especially as there is no ordering so I would hope there are no dependencies between them.

If a single analyzer has a bug we still run the others, right?